debezium-core/src/main/java/io/debezium/relational/ColumnImpl.java

```diff
@@ -133,8 +133,12 @@ public boolean isGenerated() {
     }
 
     @Override
+    public Optional<String> defaultValueExpression() {
+        return Optional.ofNullable(defaultValueExpression);
+    }
+
     public Object defaultValue() {
-        return defaultValue;
+        return null;
     }
```

Comment:
Looks like a left-over?

Also `hasDefaultValue()` should not be needed any longer; as we now can differentiate whether a default value has been provided or not by examining the `Optional` returned from `defaultValueExpression()`. The explicit `hasDefaultValue()` method was needed before as we could not tell differentiate between no default value given or a default value of `null`.

Reply:
@gunnarmorling 

The left-over has been cleaned up.

I think we should keep `hasDefaultValue` cause there are scenarios for implicit default values.

Usually, `defaultValueExpression` is null and `hasDefaultValue` is false.
But in the implicit case, `defaultValueExpression` is null but `hasDefaultValue` should be true.

In a word, the null `defaultValueExpression` does not directly indicate whether the column has a default value.

https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html

Reply:
If this scenario needs to be concerned, then the `hasDefaultValue` should also be serialized in `TableChangeSerializer`.

Reply:
I see, that's a good point. Indeed we are exporting some "implicit default values" in certain cases, and my current thinking is we probably should not do this. But it seems largely independent from the matter at hand and fixing this also would be sort of a breaking change, so it's something we should leave for Debezium 2.0 (I'll add this to the list of things to be considered for that, see [DBZ-3899](https://issues.redhat.com/browse/DBZ-3899)). No re serializing `hasDefaultValue`, I'd prefer if we'd not do that, given we should get rid of this entire implicit default value business at all. Admittedly, there'll be an inconsistency when reading back the schema history for such value, but I feel that's an acceptable limitation.

Reply:
Thanks @gunnarmorling for the detailed explanation.
I will remove all the `hasDefaultValue`.

Reply:
Ok, I'm just not sure whether we can or should do this at this point. Here's a PR where I did that (just look at the last commit): https://github.com/debezium/debezium/pull/2887. It changes some assertions, i.e. behavior exposed to consumers is slightly modified in some cases. @jpechane expressed concerns about doing this in a release like 1.8 (as opposed to a major release like 2.0). @jpechane, can you take a look at that last commit and the assertions I've changed; as stated below, I think one could also argue that the previous behavior actually was incorrect when it comes to those implicit default values.